crates/sync/anvil/src/tracker.rs

```diff
@@ -215,6 +215,20 @@ async fn process(ctx: Ctx, mut block_rcv: mpsc::UnboundedReceiver<Msg>) -> Resul
             }
         }
 
+        for erc721_token_info in ctx
+            .db
+            .get_erc721_missing_metadata(ctx.chain_id)
+            .await?
+            .into_iter()
+        {
+            let token_id = erc721_token_info.token_id;
+            let address = erc721_token_info.contract;
+            let metadata = fetch_erc721_metadata(erc721_token_info, &provider).await;
+            ctx.db
+                .save_erc721_metadata(address, ctx.chain_id, token_id, metadata)
+                .await?;
+        }
```

Comment:
now that I re-read this and the db table, this could be a problem.
I expected the `erc721` database table to have 1 record per collection. but you're storing 1 record per item, which will scale up these requests considerably, even though they're mostly duplicate (only the token URI differs from each one)

so I'd propose something different, which is to have 2 tables:
one for `erc721_collections` (which tracks name & symbol) and one for `erc721_metadata` (which tracks individual token_ids and their URI)

Reply:
to be clear. the problem is not so much with this anvil side, but with how the current data schema will translate to alchemy.com. we can't afford making that many redundant requests for every individual item